python/pyspark/sql/connect/dataframe.py

```diff
@@ -667,12 +668,70 @@ def schema(self) -> StructType:
         else:
             return self._schema
 
-    def explain(self) -> str:
+    def explain(
+        self, extended: Optional[Union[bool, str]] = None, mode: Optional[str] = None
```

Comment:
We could leverage keyword-only argument to simplify this for now. cc @zhengruifeng and @ueshin

Reply:
Let me take a look.

Reply:
Correct me if I am wrong: 

`keyword-only argument` means to not be compatible with PySpark API?

Reply:
Actually NVM. Let's just go ahead as is for now.